contracts/pools/PoolCollection.sol

```diff
@@ -872,6 +889,56 @@ contract PoolCollection is IPoolCollection, Owned, BlockNumber, Utils {
         cachedPoolToken.transferOwnership(address(targetPoolCollection));
     }
 
+    /**
+     * @dev reduces the trading liquidity of a given pool
+     *
+     * requirements:
+     *
+     * - the caller must be the owner of the contract
+     */
+    function reduceTradingLiquidity(Token pool, uint256 newTradingLiquidity) external onlyOwner {
+        Pool storage data = _poolStorage(pool);
+        PoolLiquidity memory prevLiquidity = data.liquidity;
+        uint256 bntTradingLiquidity = prevLiquidity.bntTradingLiquidity;
+
+        // if the requested trading liquidity is equal to or larger than the current trading liquidity - return
+        if (newTradingLiquidity >= bntTradingLiquidity) {
+            return;
+        }
+
+        bytes32 contextId = keccak256(abi.encodePacked(msg.sender, pool, newTradingLiquidity));
+
+        // if the requested trading liquidity is zero - reset the current trading liquidity and return
+        if (newTradingLiquidity == 0) {
+            _resetTradingLiquidity(contextId, pool, data, TRADING_STATUS_UPDATE_ADMIN);
+
+            return;
+        }
+
+        // if the requested trading liquidity is smaller than the funding limit - revert (the funding limit should be
+        // decreased below the new value first)
+        if (newTradingLiquidity < _networkSettings.poolFundingLimit(pool)) {
+            revert FundingLimitTooHigh();
+        }
+
+        // deplete the extra amount of funding currently available
+        uint256 renouncedAmount = bntTradingLiquidity - newTradingLiquidity;
+        _bntPool.renounceFunding(contextId, pool, renouncedAmount);
```

Comment:
this will reduce the current funding, while we want to reduce the trading liquidity and deplete the available funding - we actually need to request the entire available funding and burn the BNT, which will result in the available funding being equal to 0. The idea is that no further funding can be made after the reduction